### PR TITLE
RDK-44657 - Miracast merge changes to sprint/release branch

### PR DESCRIPTION
**Pull request by yuvaramachandran-gurusamy**
RDK-44657 - Miracast merge changes to sprint/release branch

### CODE REVIEW

**Review by rajnishdey — APPROVED** [review]
Looks good.

File: Miracast/MiracastPlayer/MiracastGstPlayer.cpp [hunk 1]
```diff
@@ -0,0 +1,932 @@
...
+
+void MiracastGstPlayer::destroyInstance()
+{
+    MIRACASTLOG_TRACE("Entering...");
+    if (mMiracastGstPlayer != nullptr)
+    {
+        mMiracastGstPlayer->stop();
+        if (mMiracastGstPlayer->stop())
+        {
+            MIRACASTLOG_ERROR("Stop miracast player");
```
Comment (apatel859) [inline Miracast/MiracastPlayer/MiracastGstPlayer.cpp:51]: change to info

File: Miracast/MiracastPlayer/MiracastGstPlayer.cpp [hunk 2]
```diff
@@ -0,0 +1,932 @@
...
+void MiracastGstPlayer::print_pipeline_state(GstElement *pipeline)
+{
+    MIRACASTLOG_TRACE("Entering..!!!");
+
+    if ( nullptr == pipeline )
+    {
+        pipeline = m_pipeline;
+    }
+
+    if ( nullptr == pipeline )
```
Comment (apatel859) [inline Miracast/MiracastPlayer/MiracastGstPlayer.cpp:460]: can be ombine with previous if

File: Miracast/MiracastPlayer/MiracastGstPlayer.cpp [hunk 3]
```diff
@@ -0,0 +1,932 @@
...
+    GstStructure *new_pad_struct = NULL;
+
+    MIRACASTLOG_TRACE("Entering..!!!");
+
+    if (!self->m_pipeline)
+    {
+        MIRACASTLOG_ERROR("failed to link elements!");
+        gst_object_unref(self->m_pipeline);
+        self->m_pipeline = NULL;
+        MIRACASTLOG_TRACE("Exiting..!!!");
```
Comment (apatel859) [inline Miracast/MiracastPlayer/MiracastGstPlayer.cpp:674]: combine with 671 line

File: Miracast/MiracastService/MiracastController.cpp [hunk 4]
```diff
@@ -0,0 +1,1968 @@
...
+    char sys_cls_file_ifidx[128] = {0};
+    std::string local_addr = "",
+                gw_ip_addr = "",
+                popen_buffer = "";
+    FILE *popen_file_ptr = nullptr;
+    char *current_line_buffer = nullptr;
+    std::size_t len = 0;
+    unsigned char retry_count = 3;
+
+    sprintf( sys_cls_file_ifidx , "/sys/class/net/%s/ifindex" , interface.c_str());
```
Comment (apatel859) [inline Miracast/MiracastService/MiracastController.cpp:220]: is this amlogic specifc?

File: Miracast/MiracastService/MiracastController.cpp [hunk 5]
```diff
@@ -0,0 +1,1968 @@
...
+
+    if (!ifIndexFile.good()) {
+        MIRACASTLOG_ERROR("Could not find [%s]\n",sys_cls_file_ifidx);
+        return std::string("");
+    }
+
+    sprintf(command, "/sbin/udhcpc -v -i ");
+    sprintf(command + strlen(command), interface.c_str());
+
+    if (0 == access("/opt/miracast/p2p_udhcpc.script", F_OK))
```
Comment (apatel859) [inline Miracast/MiracastService/MiracastController.cpp:232]: do we need this override?

File: Miracast/MiracastService/MiracastController.cpp [hunk 6]
```diff
@@ -0,0 +1,1968 @@
...
+
+        command = "/usr/bin/dnsmasq -p0 -i ";
+        command.append(interface.c_str());
+        command.append(" -F 192.168.59.50,192.168.59.230,255.255.255.0,24h --log-queries=extra");
+    }
+#endif
+    MIRACASTLOG_VERBOSE("command : [%s]", command.c_str());
+    system(command.c_str());
+
+#if 0
```
Comment (apatel859) [inline Miracast/MiracastService/MiracastController.cpp:354]: do we need this code?

File: Miracast/MiracastService/MiracastController.cpp [hunk 7]
```diff
@@ -0,0 +1,1968 @@
...
+                m_notify_handler->onMiracastServiceClientConnectionError( device_mac , device_name , MIRACAST_SERVICE_ERR_CODE_P2P_CONNECT_ERROR );
+            }
+        }
+    }
+    MIRACASTLOG_TRACE("Exiting...");
+    return ret;
+}
+
+#if 0
+MiracastError MiracastController::start_streaming()
```
Comment (apatel859) [inline Miracast/MiracastService/MiracastController.cpp:637]: do we    need this?

File: Miracast/MiracastService/MiracastController.cpp [hunk 8]
```diff
@@ -0,0 +1,1968 @@
...
+                                    MIRACASTLOG_ERROR("Local IP address is not obtained");
+                                }
+                                else
+                                {
+                                    if (m_groupInfo->goIPAddr.empty())
+                                    {
+                                        MIRACASTLOG_VERBOSE("default_gw_ip [%s]\n", default_gw_ip.c_str());
+                                        m_groupInfo->goIPAddr.append(default_gw_ip);
+                                    }
+                                    #ifdef NATIVE_PLAYER_INTEGRATED
```
Comment (apatel859) [inline Miracast/MiracastService/MiracastController.cpp:1050]: when this flag will be enabled or disabled

File: Miracast/MiracastService/MiracastService.cpp [hunk 9]
```diff
@@ -0,0 +1,1043 @@
...
+				{
+					set_loglevel(level);
+				}
+			}
+
+			MIRACASTLOG_INFO("Exiting..!!!");
+			returnResponse(success);
+		}
+
+#ifdef ENABLE_MIRACAST_SERVICE_TEST_NOTIFIER
```
Comment (apatel859) [inline Miracast/MiracastService/MiracastService.cpp:640]: are we enabling this flag for release build?

File: Miracast/P2P/MiracastP2P.cpp [hunk 10]
```diff
@@ -0,0 +1,546 @@
...
+        wpa_ctrl_close(m_wpa_p2p_cmd_ctrl_iface);
+        m_wpa_p2p_cmd_ctrl_iface = nullptr;
+        MIRACASTLOG_TRACE("Exiting...");
+        return MIRACAST_P2P_INIT_FAILED;
+    }
+    MIRACASTLOG_INFO("WIFI_HAL: m_wpa_p2p_ctrl_monitor created successfully.");
+    if (wpa_ctrl_attach(m_wpa_p2p_ctrl_monitor) != 0)
+    {
+        MIRACASTLOG_ERROR("WIFI_HAL: p2p wpa_ctrl_attach failed ");
+        wpa_ctrl_close(m_wpa_p2p_cmd_ctrl_iface);
```
Comment (apatel859) [inline Miracast/P2P/MiracastP2P.cpp:188]: we can write a utlity function to avoid repeataion

File: Miracast/common/MiracastCommon.cpp [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+                free(data_ptr);
+            }
+        }
+    }
+    MIRACASTLOG_TRACE("Exiting...");
+    return status;
+}
+
+#if 0
+std::string parse_opt_flag( std::string file_name , bool integer_check )
```
Comment (apatel859) [inline Miracast/common/MiracastCommon.cpp:151]: do we  need this

File: Miracast/MiracastPlayer/MiracastGstPlayer.cpp (hunk 1 above)
Comment (yuvaramachandran-gurusamy) [reply Miracast/MiracastPlayer/MiracastGstPlayer.cpp]: Addressed in https://github.com/rdkcentral/rdkservices/pull/4618/commits/abb5db62d2585d5e30a40d8f7c1645336732c0ea

File: Miracast/MiracastPlayer/MiracastGstPlayer.cpp (hunk 2 above)
Comment (yuvaramachandran-gurusamy) [reply Miracast/MiracastPlayer/MiracastGstPlayer.cpp]: Addressed in https://github.com/rdkcentral/rdkservices/pull/4618/commits/abb5db62d2585d5e30a40d8f7c1645336732c0ea

File: Miracast/MiracastPlayer/MiracastGstPlayer.cpp (hunk 3 above)
Comment (yuvaramachandran-gurusamy) [reply Miracast/MiracastPlayer/MiracastGstPlayer.cpp]: Addressed in https://github.com/rdkcentral/rdkservices/pull/4618/commits/abb5db62d2585d5e30a40d8f7c1645336732c0ea

File: Miracast/MiracastService/MiracastController.cpp (hunk 4 above)
Comment (yuvaramachandran-gurusamy) [reply Miracast/MiracastService/MiracastController.cpp:220]: No. It is generic linux check to whether iface created or not

File: Miracast/MiracastService/MiracastController.cpp (hunk 5 above)
Comment (yuvaramachandran-gurusamy) [reply Miracast/MiracastService/MiracastController.cpp]: It is not required. I removed it as part of https://github.com/rdkcentral/rdkservices/pull/4618/commits/019f3d11389a6271609f0a3153b3a3984f89ca64

File: Miracast/MiracastService/MiracastController.cpp (hunk 6 above)
Comment (yuvaramachandran-gurusamy) [reply Miracast/MiracastService/MiracastController.cpp]: It is not required. I removed it as part of https://github.com/rdkcentral/rdkservices/commit/019f3d11389a6271609f0a3153b3a3984f89ca64

File: Miracast/MiracastService/MiracastController.cpp (hunk 7 above)
Comment (yuvaramachandran-gurusamy) [reply Miracast/MiracastService/MiracastController.cpp]: It is not required. I removed it as part of https://github.com/rdkcentral/rdkservices/commit/019f3d11389a6271609f0a3153b3a3984f89ca64

File: Miracast/MiracastService/MiracastController.cpp (hunk 8 above)
Comment (yuvaramachandran-gurusamy) [reply Miracast/MiracastService/MiracastController.cpp]: It flag will not be enabled. I removed it as part of https://github.com/rdkcentral/rdkservices/commit/019f3d11389a6271609f0a3153b3a3984f89ca64

File: Miracast/MiracastService/MiracastService.cpp (hunk 9 above)
Comment (yuvaramachandran-gurusamy) [reply Miracast/MiracastService/MiracastService.cpp:640]: Currently It is enabled. Now disabled it as part of https://github.com/rdkcentral/rdkservices/pull/4618/commits/56e11f35b7a6a7ea1fcc73da62a69f46ef3663d5 and https://github.com/rdkcentral/rdkservices/pull/4618/commits/af878fc3ccc6c61fff851af1f44dde68b5b892e3

File: Miracast/common/MiracastCommon.cpp (hunk 11 above)
Comment (yuvaramachandran-gurusamy) [reply Miracast/common/MiracastCommon.cpp:151]: This API generic to read the opt flags. So we can keep it in this common file. Later we can use it

File: Miracast/P2P/MiracastP2P.cpp (hunk 10 above)
Comment (yuvaramachandran-gurusamy) [reply Miracast/P2P/MiracastP2P.cpp]: Addressed the review comments in https://github.com/rdkcentral/rdkservices/pull/4618/commits/9e283d757407fc8e7010eaed21427ea2af2865d6

**Review by apatel859 — APPROVED** [review]

**Pull request merged**